gammapy/datasets/spectrum.py

```diff
@@ -336,23 +336,60 @@ def energy_range(self):
 
         return u.Quantity([e_min.min(), e_max.max()])
 
-    def plot_fit(self):
-        """Plot counts and residuals in two panels.
+    def plot_fit(self, ax_spectrum=None, ax_residuals=None, kwargs_spectrum=None, kwargs_residuals=None):
+        """Plot spectrum and residuals in two panels.
 
-        Calls ``plot_counts`` and ``plot_residuals``.
+        Calls `~SpectrumDataset.plot_excess` and `~SpectrumDataset.plot_residuals`.
+
+        Parameters
+        ----------
+        ax_spectrum : `~matplotlib.axes.Axes`
+            Axes to plot spectrum on.
+        ax_residuals : `~matplotlib.axes.Axes`
+            Axes to plot residuals on.
+        kwargs_spectrum : dict
+            Keyword arguments passed to `~SpectrumDataset.plot_excess`.
+        kwargs_residuals : dict
+            Keyword arguments passed to `~SpectrumDataset.plot_residuals`.
+
+        Returns
+        -------
+        ax_spectrum, ax_residuals : `~matplotlib.axes.Axes`
+            Spectrum and residuals plots.
         """
-        from matplotlib.gridspec import GridSpec
         import matplotlib.pyplot as plt
+        from matplotlib.gridspec import GridSpec
+
+        if not ax_spectrum or not ax_residuals:
```

Comment:
I think we could possibly even simplify and require that `ax_spectrum` and `ax_residuals` must be both provided, and just raise an error if this is not the case...